web/ui/src/Services/AppsController.js

```diff
@@ -292,34 +267,108 @@
                 $scope.templates.data = utils.mapToArr(templates);
             });
         }
-       
-        function fetchInternalServiceInstances(app) {
-            return resourcesFactory.v2.getInternalServiceInstances(app.id).then( data => {
-                app.instances = data;
+
+        function refreshApps() {
+            return fetchApps()
+                .then(fillOutApps)
+                .then(fetchServicesHealth)
+                .then(fetchInternalServicesHealth)
+                .then(() => touch());
+        }
+
+        function fetchApps() {
+            return $q.all([fetchInternalService(), fetchServices()])
+                .then(results => {
+                    $scope.apps = [];
+
+                    var isvc = results[0];
+                    $scope.apps.push(isvc);
+
+                    var services = results[1];
+                    services.forEach(s => {
+                        $scope.apps.push(s);
+                    });
+                });
+        }
+
+        function fetchChildEndpoints(app) {
+            return resourcesFactory.v2.getServiceChildPublicEndpoints(app.id).then(data => {
+                app.publicEndpoints = data;
             });
         }
 
-        function getInternalServicesHealth() {
+        function fillOutApps() {
+            let promises = [];
+            $scope.apps.forEach(a => {
+                if (a.isIsvc()) {
+                    promises.push(() => a.fetchInstances());
+                } else {
+                    promises.push(fetchChildEndpoints(a));
+                }
+            });
+
+            return $q.all(promises);
+        }
+
+        function fetchServices() {
+            let deferred = $q.defer();
+            resourcesFactory.v2.getTenants().then( data => {
+                var services = data.map(s => new Service(s));
+                deferred.resolve(services);
+                },
+                error => {
+                    console.warn(error);
+                    deferred.reject();
+                });
+            return deferred.promise;
+        }
+
+        function fetchServicesHealth() {
+            if (!$scope.apps) { return; }
+
+            $scope.apps.forEach(app => {
+                if (!app.isIsvc()) {
+                    app.fetchAllStates();
+                }
+            });
+        }
+
+        function fetchInternalService() {
+            let deferred = $q.defer();
+            resourcesFactory.v2.getInternalServices()
+                .then(data => {
+                    let parentIndex = data.findIndex(i => !i.Parent);
+                    if (parentIndex > -1) {
+                        var internalServices = new InternalService(data[parentIndex]);
+                        internalServices.fetchInstances();
```

Comment:
This call was actually unnecessary.  The fetchInstances call has been moved into a promise chain.